perf/run-perf.sh

```diff
@@ -0,0 +1,221 @@
+    if [ ! -d $OPENSSL_DIR ] ; then
+        echo "OpenSSL $VERSION not found ($OPENSSL_DIR does not exit"
+        exit 1
+    fi
+done
+
+
+#
+# directory where perf tools sources live
+#
+# TOOLS_PATH=~/work.openssl/tools.sashan/perf
+#
+if [ -z ${TOOLS_PATH+x} ] ; then
+    echo "TOOLS_PATH env. variable not set"
+    exit 1
+fi
+
+
+#
+# directory where clone of github repository lives
+# handshake test (et. al.) use certificates from
+# there.
+#
+# OPENSSL_SRC=~/work.openssl/openssl/
+#
+if [ -z ${OPENSSL_SRC+x} ] ; then
+    echo "OPENSSL_SRC env. variable not set"
+    exit 1
+fi
+CERT_DIR=$OPENSSL_SRC/test/certs
+if [ ! -d $CERT_DIR ] ; then
+    echo "$CERT_DIR not found in $OPENSSL_SRC"
+    exit 1
+fi
+
+#
+# directory where to store results
+#
+RESULTS=results
+#
+# Iterations to run each tool
+#
+ITERATIONS=25
+
+function openssl.env {
+    OPENSSL_VERSION=$1;
+    OPENSSL_DIR=$OPENSSL_BINARIES/openssl-$OPENSSL_VERSION
+    if [ -d $OPENSSL_DIR ] ; then
+        #
+        # OpenBSD build (and perhaps other place 64-bit version
+        # to prefix/lib. while for example Debian build installs
+        # openssl under prefix/lib64
+        #
+        if [ -d $OPENSSL_DIR/lib ] ; then
+            LD_LIBRARY_PATH=$OPENSSL_DIR/lib
+            # set them for perf tools testing
+            TARGET_OSSL_LIBRARY_PATH=$OPENSSL_DIR/lib
+            TARGET_OSSL_INCLUDE_PATH=$OPENSSL_DIR/include
+            OPENSSL_HEADERS=$OPENSSL_DIR/include
+            OPENSSL_LIB_PATH=$OPENSSL_DIR/lib
+        elif [ -d $OPENSSL_DIR/lib64 ] ; then
+            LD_LIBRARY_PATH=$OPENSSL_DIR/lib64
+            # set them for perf tools testing
+            TARGET_OSSL_LIBRARY_PATH=$OPENSSL_DIR/lib64
+            TARGET_OSSL_INCLUDE_PATH=$OPENSSL_DIR/include
+            OPENSSL_HEADERS=$OPENSSL_DIR/include
+            OPENSSL_LIB_PATH=$OPENSSL_DIR/lib64
+        else
+            echo "$OPENSSL_DIR does not exist"
+            exit 1
+        fi
+    else
+        echo "$OPENSSL_DIR does not exist"
+        exit 1
+    fi
+}
+
+mkdir -p $RESULTS
+rm -rf $RESULTS/*
+
+#
+# execute all tools we have to convey performance test.
+# each program we run prints single number: microseconds
+# it took to run it.
+#
+for TOOL in evp_fetch randbytes handshake sslnew newrawkey rsasign \
+        x509storeissuer providerdoall rwlocks-rlock rwlocks-wlock \
+	pkeyread-dh-der pkeyread-dhx-der pkeyread-dsa-der pkeyread-ec-der \
+	pkeyread-rsa-der pkeyread-x25519-der pkeyread-dh-pem pkeyread-dhx-pem \
+	pkeyread-dsa-pem pkeyread-ec-pem pkeyread-rsa-pem \
+	pkeyread-x25519-pem ; do
+    CMD=$TOOLS_PATH/$TOOL
+
+    for VERSION in master 3.0 1.1.1 ; do
+        pushd `pwd`
+        cd $TOOLS_PATH
+        openssl.env $VERSION;
+        make clean
+        if [ "$VERSION" == "1.1.1" ] ; then
+            make all111
+        else
+            make
+        fi
+        popd
+        for THREADS in 1 2 4 8 16 32 64 ; do
+            if [[ ! -x $CMD ]] ; then 
```

Comment:
Assuming `$CMD` is in the format of `path/tool some args` you could just use `-x $(echo $CMD | cut -f 1 -d ' ')`

Reply:
currently `$CMD` does not hold arguments. it's just file path (see line 121 above where `$CMD` is set). The assumption does not hold. Not sure where this comment is heading too.

Reply:
I'm keeping this test. the test checks if particular tool `$CMD` exists. For example we don't build `evp_fetch` for openssl-1.1.1. For tools which don't exist we put N/A as a result.